### PR TITLE
[external-dns] chore: change chart location

### PR DESCRIPTION
**Pull request by alejandroEsc**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-72458

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: addons/external-dns/external-dns.yaml [hunk 1]
```diff
@@ -24,7 +24,8 @@ spec:
     - name: none
       enabled: false
   chartReference:
-    chart: stable/external-dns
+    repo: https://charts.bitnami.com/bitnami
+    chart: bitnami
```
Comment (gracedo) [inline addons/external-dns/external-dns.yaml:28]:
```suggestion
    chart: bitnami/external-dns
```

File: addons/external-dns/external-dns.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     kubeaddons.mesosphere.io/name: external-dns
   annotations:
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.7.0-1"
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.7.0-2"
     appversion.kubeaddons.mesosphere.io/external-dns: "0.7.0"
     values.chart.helm.kubeaddons.mesosphere.io/external-dns: "https://raw.githubusercontent.com/helm/charts/b9191a5b526d4988b7938be2049370a1687037a0/stable/external-dns/values-production.yaml"
```
Comment (gracedo) [inline addons/external-dns/external-dns.yaml:11]:
```suggestion
    values.chart.helm.kubeaddons.mesosphere.io/external-dns: "https://raw.githubusercontent.com/bitnami/charts/9eca03cb2f4b9d4a5b7e877711f810f9151ea2a3/bitnami/external-dns/values.yaml"
```
(I think that's the right values for 2.20.4?)

**Review by gracedo — APPROVED** [review]

**Review by shaneutt — APPROVED** [review]
LGTM, but before we merge this let's please do a manual check just to make sure the chart is behaving exactly as we intend.

**Review by makkes — APPROVED** [review]

**Comment by alejandroEsc** [comment]
The TC test passed, though it's not reporting.

**Pull request merged**